scripts/sample_json_generator.py

```
@@ -0,0 +1,611 @@
+        
+        Raises:
+            ValueError: If fetching the schema from a URL fails.
+            FileNotFoundError: If the specified local schema file does not exist.
+        """
+        if path_or_url.startswith(("http://", "https://")):
+            try:
+                response = requests.get(path_or_url, timeout=15)
+                response.raise_for_status()
+                return response.json()
+            except requests.RequestException as e:
+                raise ValueError(f"Failed to fetch schema: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)